### PR TITLE
Removal of installation extras in setup.py

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Remove extra dependencies to install backends from setup.py
Add [plots] extra dependencies to install optional packages to visualize plots.  

### Related tickets

134503

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
nightly/job/install_onnx/439/
nightly/job/install_ov/468/
nightly/job/install_pt_cpu/463/
nightly/job/install_pt_gpu/468/
nightly/job/ubuntu20_install_tf/574/

File: nncf/common/utils/decorators.py [hunk 1]
```diff
@@ -39,8 +39,7 @@ def wrapped_f(*args, **kwargs):
                     nncf_logger.warning(
                         f"{ex.msg} Please install NNCF package with dev "
                         "extra. Use one of the following commands "
-                        '"pip install .[dev]" running from the repository '
-                        'root directory or "pip install nncf[dev]"'
+                        '"pip install -r requirements-dev.txt" running from the repository.'
```
Comment (alexsu52) [inline nncf/common/utils/decorators.py:42]: I wonder how the user will do this if he installed nncf from pip. Could you save the behavior from develop for this case?

Comment (AlexanderDokuchaev) [reply nncf/common/utils/decorators.py]: return nncf[dev]

File: nncf/common/utils/decorators.py [hunk 2]
```diff
@@ -39,8 +39,8 @@ def wrapped_f(*args, **kwargs):
                     nncf_logger.warning(
                         f"{ex.msg} Please install NNCF package with dev "
```
Comment (alexsu52) [inline nncf/common/utils/decorators.py:40]:
```suggestion
                        f"{ex.msg} Please install NNCF package with plots "
```

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**